shopBuddy/shopBuddy/VCs/AddItemViewController.m

```diff
@@ -35,19 +36,35 @@ - (IBAction)didTapScanBarcode:(id)sender {
 }
 
 - (IBAction)didTapSearch:(id)sender {
-    //TODO: Search field validation
+    __weak __typeof__(self) weakSelf = self;
     [[BarcodeAPIManager shared] searchItemsWithQuery:_searchField.text completion:^(NSMutableArray<Item*> *items, NSError *error) {
-        //TODO: handle error
-        self->_searchResults = items;
-        [self->_tableView reloadData];
+        if(items) {
+            __strong __typeof(weakSelf)strongSelf = weakSelf;
+            if(strongSelf)
+            {
+                strongSelf->_searchResults = items;
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    [strongSelf->_tableView reloadData];
+                });
+            }
+        }
+        else {
+            UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Item Not Found" message:@"The searched item is not found." preferredStyle:(UIAlertControllerStyleAlert)];
+            UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action){}];
+            [alert addAction:okAction];
+            [weakSelf presentViewController:alert animated:YES completion:^{
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    [self.navigationController popToRootViewControllerAnimated:YES];
```

Comment:
Should not use self inside of async block. Async blocks not only include network requests but also blocks that run in a separate thread/queue like the ones scheduled using dispatch_async.